central/graphql/resolvers/image_components.go

```diff
@@ -105,24 +155,256 @@ func (resolver *Resolver) ImageComponentCount(ctx context.Context, args RawQuery
+	}
+
+	q := search.EmptyQuery()
+	q.Pagination = &v1.QueryPagination{
+		Limit:  1,
+		Offset: 0,
+		SortOptions: []*v1.QuerySortOption{
+			{
+				Field:    search.ImageScanTime.String(),
+				Reversed: true,
+			},
+		},
+	}
+
+	images, err := imageLoader.FromQuery(resolver.withImageComponentScope(ctx), q)
+	if err != nil || len(images) == 0 {
+		return nil, err
+	} else if len(images) > 1 {
+		return nil, errors.New("multiple images matched for last scanned image component query")
+	}
+
+	return timestamp(images[0].GetScan().GetScanTime())
+}
+
+// Location returns the location of the component.
+func (resolver *imageComponentResolver) Location(ctx context.Context, args RawQuery) (string, error) {
+	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.ImageComponents, "Location")
+	var imageID string
+	scope, hasScope := scoped.GetScope(ctx)
+	if hasScope && scope.Level == v1.SearchCategory_IMAGES {
+		imageID = scope.ID
+	} else if !hasScope || scope.Level != v1.SearchCategory_IMAGES {
+		var err error
+		imageID, err = getImageIDFromIfImageShaQuery(ctx, resolver.root, args)
+		if err != nil {
+			return "", errors.Wrap(err, "could not determine component location")
+		}
+	}
+
+	if imageID == "" {
+		return "", nil
+	}
+
+	query := search.NewQueryBuilder().AddExactMatches(search.ImageSHA, imageID).AddExactMatches(search.ComponentID, resolver.data.GetId()).ProtoQuery()
+	edges, err := resolver.root.ImageComponentEdgeDataStore.SearchRawEdges(ctx, query)
+	if err != nil || len(edges) == 0 {
+		return "", err
+	}
+	return edges[0].GetLocation(), nil
+}
+
+// PlottedImageVulnerabilities returns the data required by top risky entity scatter-plot on vuln mgmt dashboard
+func (resolver *imageComponentResolver) PlottedImageVulnerabilities(ctx context.Context, args RawQuery) (*PlottedImageVulnerabilitiesResolver, error) {
+	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.ImageComponents, "PlottedImageVulnerabilities")
+	return resolver.root.PlottedImageVulnerabilities(resolver.withImageComponentScope(ctx), args)
+}
+
+func (resolver *imageComponentResolver) TopImageVulnerability(ctx context.Context) (ImageVulnerabilityResolver, error) {
+	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.ImageComponents, "TopImageVulnerability")
+	return resolver.root.TopImageVulnerability(resolver.withImageComponentScope(ctx), RawQuery{})
+}
+
+func (resolver *imageComponentResolver) ID(_ context.Context) graphql.ID {
+	return graphql.ID(resolver.data.GetId())
+}
+
+func (resolver *imageComponentResolver) LayerIndex() *int32 {
+	return nil
+}
+
+func (resolver *imageComponentResolver) UnusedVarSink(ctx context.Context, args RawQuery) *int32 {
+	return nil
+}
+
+// Follows are functions that return information that is nested in the CVEInfo object
+// or are convenience functions to allow time for UI to migrate to new naming schemes
+
+func (resolver *imageComponentResolver) FixedIn(_ context.Context) string {
```

Comment:
I see `fixedIn` is deprecated in favor of `fixedBy` on `ln31` . Add `fixedBy` sub resolver to the `ImageComponent` schema type.

Reply:
Yeah, I was going to leave the resolver in with a deprecation on the graphql schema since it's part of the interface for the time being